applications/GeoMechanicsApplication/tests/cpp_tests/custom_workflows/test_time_loop_executor.cpp

```diff
@@ -335,43 +317,57 @@ KRATOS_TEST_CASE_IN_SUITE(ExpectFinalizeSolutionStepCalledOnceForEveryStep, Krat
-    executor.SetSolverStrategyWrapper(solver_strategy);
+    auto p_solver_strategy = std::make_shared<MockSolverStrategy>();
+    executor.SetSolverStrategyWrapper(p_solver_strategy);
+
+    EXPECT_CALL(*p_solver_strategy, FinalizeSolutionStep).Times(2);
     const auto step_states = executor.Run(TimeStepEndState{});
-    KRATOS_EXPECT_EQ(step_states.size(), solver_strategy->GetCountFinalizeSolutionStepCalled());
+    KRATOS_EXPECT_EQ(2, step_states.size());
 }
 
 KRATOS_TEST_CASE_IN_SUITE(ExpectOutputIsInitializedAndFinalizedWhenRunCompletesOk, KratosGeoMechanicsFastSuiteWithoutKernel)
 {
     TimeLoopExecutor executor;
     const auto       wanted_num_of_cycles_per_step = std::size_t{1};
     executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(wanted_num_of_cycles_per_step));
-    auto solver_strategy = std::make_shared<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged);
-    executor.SetSolverStrategyWrapper(solver_strategy);
+    auto p_solver_strategy = std::make_shared<MockSolverStrategy>();
+    executor.SetSolverStrategyWrapper(p_solver_strategy);
+
+    EXPECT_CALL(*p_solver_strategy, InitializeOutput).Times(1);
+    EXPECT_CALL(*p_solver_strategy, FinalizeOutput).Times(1);
     executor.Run(TimeStepEndState{});
-    KRATOS_EXPECT_EQ(solver_strategy->GetCountInitializeOutputCalled(), 1);
-    KRATOS_EXPECT_EQ(solver_strategy->GetCountFinalizeOutputCalled(), 1);
 }
 
 KRATOS_TEST_CASE_IN_SUITE(ExpectOutputIsInitializedAndFinalizedWhenRunThrows, KratosGeoMechanicsFastSuiteWithoutKernel)
 {
     TimeLoopExecutor executor;
     const auto       wanted_num_of_cycles_per_step = std::size_t{1};
     executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(wanted_num_of_cycles_per_step));
-    auto solver_strategy = std::make_shared<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged);
-    solver_strategy->SetOutputProcessCallback([]() { throw Exception{"Test exception"}; });
-    executor.SetSolverStrategyWrapper(solver_strategy);
-
-    auto exception_caught = false;
-    try {
-        executor.Run(MakeConvergedStepState());
-    } catch (const Exception&) {
-        exception_caught = true;
-    }
+    auto p_solver_strategy = std::make_shared<MockSolverStrategy>();
+    p_solver_strategy->SetOutputProcessCallback([]() { throw Exception{"Test exception"}; });
+    executor.SetSolverStrategyWrapper(p_solver_strategy);
+
+    EXPECT_CALL(*p_solver_strategy, InitializeOutput).Times(1);
+    EXPECT_CALL(*p_solver_strategy, FinalizeOutput).Times(1);
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(executor.Run(MakeConvergedStepState()), "Test exception");
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TimeLoopExecutor_CallsProcessExecuteBeforeSolutionLoop_AfterInitialize,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    TimeLoopExecutor executor;
+    const auto       increments = std::vector<double>{1.1, 1.2, 1.3, 1.4};
```

Comment:
The template parameter can be deduced from the initializer list:

```suggestion
    const auto       increments = std::vector{1.1, 1.2, 1.3, 1.4};
```

Reply:
In the end I passed an empty list, since that is enough for this test (we don't need any time steps), but removed it in the other places (will double check whether all compilers we use like it though)